core-services/src/main/java/org/zalando/nakadi/repository/kafka/KafkaRecordDeserializer.java

```diff
@@ -93,4 +116,10 @@ private JSONObject mapToJson(final Metadata metadata) {
         }
         return metadataObj;
     }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class MetadataHolder {
+        @JsonSetter("metadata")
+        private JsonNode metadata;
+    }
```

Comment:
I tried to look into the jackson databind source and it would take me too long to figure out exactly how the parsing works but [this piece of code](https://github.com/FasterXML/jackson-databind/blob/9d18eb057b692819b7dc40f87740cbc8707f89a2/src/main/java/com/fasterxml/jackson/databind/deser/std/StdNodeBasedDeserializer.java#L98) doesn't give me much hope that it bears any benefit compared to `.readTree()`.

If we strictly wanted to be sure that we don't allocate or parse things unnecessarily we could use the Jackson streaming API which is used by databind under the hood ([example code](http://www.cowtowncoder.com/blog/archives/2009/01/entry_132.html)), but I think it's a bit of an overkill.